### PR TITLE
doc: Fix minimum Postgres version requirement

### PR DESCRIPTION
**Pull request by jseldess**
The min Postgres version in our Postgres/Debezium guide is out-of-date. The PR fixes that. It also make the minimum requirement language more explicit in other guides.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by morsapaes — APPROVED** [review]
Should it also be mentioned in the [reference page](https://materialize.com/docs/sql/create-source/postgres/)? The guides feel like the most relevant spot to add this, so we could go either way.

**Comment by jseldess** [comment]
TFTR, @morsapaes. Added info to the `CREATE SOURCE` page.

File: doc/user/content/sql/create-source/postgres.md [hunk 1]
```diff
@@ -12,14 +12,14 @@ aliases:
   - /sql/create-source/postgresql
 ---
 
+{{% create-source/intro %}}
+Materialize supports PostgreSQL 11+ as a data source. To connect to a PostgreSQL instance, you first need to [create a connection](#creating-a-connection) that specifies access and authentication parameters. Once created, a connection is **reusable** across multiple `CREATE SOURCE` statements.
```
Comment (morsapaes) [inline doc/user/content/sql/create-source/postgres.md:16]:
```suggestion
To connect to a PostgreSQL (11+) instance, you first need to [create a connection](#creating-a-connection) that specifies access and authentication parameters. Once created, a connection is **reusable** across multiple `CREATE SOURCE` statements.
```

Comment (morsapaes) [inline doc/user/content/sql/create-source/postgres.md:16]:
```suggestion
Materialize supports PostgreSQL (11+) as a data source. To connect to a PostgreSQL instance, you first need to [create a connection](#creating-a-connection) that specifies access and authentication parameters. Once created, a connection is **reusable** across multiple `CREATE SOURCE` statements.
```

**Review by morsapaes — COMMENTED** [review]
Suggestion to make it more understated. Feel free to merge as you please!

**Pull request merged**